<a href="https://colab.research.google.com/github/SwithJonlamonthon/BUcode/blob/main/Recursive_Swith_Jonlamonthon_6630900022.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

1. Show step-by-step to find the returned values of the algorithm quiz1.

In [ ]:
def q(n):
    if(n<6):
      return 4*n
    else:
      return (3*q(n-6)+2)
print(q(5))
print(q(10))
print(q(16))



20
50
152


>1.1 

---


    1-> n = 5 ; if n < 6 //True ; return 4 * n ; Ans = 20
Ans = 20
>1.2

---


    1->n = 10 ; if n < 6 //False ; return 3*q(n-6)+2; n = 4
    2->n = 4 ; if n < 6 //True ; return 4 * n; n = 16
    ans = 3*16+2 = 50
Ans = 50
>1.3

---
    1->n = 16 ; if n < 6 //False ; return 3*q(n-6)+2; n = 10
    2->n = 10 ; if n < 6 //True ; return 3*q(n-6)+2; n = 4
    3->2->n = 4 ; if n < 6 //True ; return 4 * n; n = 16
    Ans ->1-> 3*16+2 -> return 50
          2-> 3*50+2 -> Ans = 152
Ans = 152












2. Show step-by-step to find the returned values of the algorithm quiz2`

In [11]:
def q2(x,y):
      if(x < y):
        return -5
      else:
        return (q2((x-y),y+2)+y)  
print(q2(2,8))
print(q2(8,2))
print(q2(25,12))

5
11
17


2.1

---


    1-> x = 2 , y = 8 ; if (x < y) //True ; return -5
  Ans = -5

2.2

---

    1-> x = 8 , y = 2 ; if (x < y) //False ; return (q2((x-y),y+2)+y) ; x = 6, y = 4
    2-> x = 6, y = 4 ; if (x < y) //False ; return (q2((x-y),y+2)+y) ; x = 2, y = 6 
    3-> x = 2 , y = 6 ; if (x < y) //True ; return -5
    Ans ->1 -> -5 + 4 //Cause the parameter from 2ed round have change(y+2)
          2 -> -1 + 2 //1st round have y = 2 
    
Ans = 1

2.3

---

    1-> x = 25 , y = 12 ; if (x < y) //False ; return (q2((x-y),y+2)+y) ; x = 13, y = 14
    2-> x = 13 , y = 14 ; if (x < y) //True ; return -5 ; 
    Ans ->1-> -5 + 12 

Ans = 7



The combination of n objects taken by k at a time can be determined according to the recursive algorithm below.

In [19]:
def com(n,k):
    if k == 0 or n == k:
      return 1
    elif n > k and k > 0:
      return com(n-1,k)+com(n-1,k-1)

print(com(3,2))



3


if n = k or k = 0 this function gonna return 1 

---
if n < k this function gonna return NULL

---
IF n = 3 k = 2

---
    1-> if(k == 0 or n == k)//false/ else  if(n > k and k > 0)//True
        return com(n-1,k)+com(n-1,k-1) 
        1.1 n = 2 , k = 2 / 1.2 n = 2 , k = 1
    2-> if(k == 0 or n == k)//1.1 true/1.2 false/ else  if(n > k and k > 0)//1.2 True/ 1.1 false
        1.1 return 1(This function stop) / 1.2 return com(n-1,k)+com(n-1,k-1) 
          1.2.1 n = 1 , k = 1 / 1.2.2 n = 1 , k = 0
    3-> 1.2.1 and 1.2.2 return 1 because if(k == 0 or n == k)// True

    Ans-> 1 -> 1 + 1 
          2 -> 1 + 2

Ans = 3

        

